service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/ControlledAiNotebookHandler.java

```diff
@@ -53,4 +57,20 @@ public WsmResource makeResourceFromDb(DbResource dbResource) {
             .build();
     return resource;
   }
+
+  /**
+   * Generate Ai notebook cloud instance id that meets the requirements for a valid instance id.
+   *
+   * <p>Instance name id must match the regex '(?:[a-z](?:[-a-z0-9]{0,63}[a-z0-9])?)', i.e. starting
+   * with a lowercase alpha character, only alphanumerics and '-' of max length 63. I don't have a
+   * documentation link, but gcloud will complain otherwise.
+   */
+  public String generateCloudName(@Nullable UUID workspaceUuid, String aiNotebookName) {
```

Comment:
can you add unit test for the three handler please? I think we should have test cases coverage for the edge cases. thanks

Reply:
Added. Thank you for the confirming, yu.